### PR TITLE
Avoid uploading eslint sarif for dependabot PR

### PR DESCRIPTION
**Pull request by aeisenberg**
Dependabot does not have `security-events: write` permission.

See https://github.com/github/codeql-action/pull/2485

### CODE REVIEW

**Review by angelapwen — DISMISSED** [review]
Thanks!!

File: .github/workflows/pr-checks.yml [hunk 1]
```diff
@@ -30,7 +30,7 @@ jobs:
       - name: Upload sarif
         uses: github/codeql-action/upload-sarif@v3
         # Only upload SARIF for the latest version of Node.js
-        if: "always() && matrix.node-types-version == 'current'"
+        if: "always() && matrix.node-types-version == 'current' && !startsWith(github.head_ref, 'dependabot/')"
```
Comment (aeisenberg) [inline .github/workflows/pr-checks.yml:33]: Don't attempt to upload during cancelled jobs.

```suggestion
        if: "!cancelled() && matrix.node-types-version == 'current' && !startsWith(github.head_ref, 'dependabot/')"
```

**Comment by aeisenberg** [comment]
Minor change to align with what we're doing for vscode-codeql. Can you re-review?

**Review by angelapwen — APPROVED** [review]
Makes sense!

**Pull request merged**